core/api/src/app/wallets/register-broadcasted-payout-txn.ts

```diff
@@ -21,6 +21,9 @@ export const registerBroadcastedPayout = async ({
   const { estimatedProtocolFee } = setTxIdResult
   if (estimatedProtocolFee.amount === proportionalFee.amount) return true
 
+  const isRecorded = await LedgerFacade.isOnChainFeeReconciliationRecorded(payoutId)
+  if (isRecorded !== false) return isRecorded
```

Comment:
this is a weird validation.. is it to avoid error validation?

Reply:
Nope, it's that if there was an error we should return the error here, or if the onchain reconciliation was already recorded (`true`) we should also return here. We would only continue if the `isOnChainReconciliationrRecorded` is `false` in which case the rest of the function continues on to record the reconciliation.

Reply:
yep, I meant that `!isRecorded || isRecorded instanceof Error`